src/lidar/read_rp_lidar.cpp

```diff
@@ -12,60 +12,84 @@
 
 using namespace navtypes;
 
-std::atomic<bool> lidar_initialized(false);
 constexpr uint32_t RPLIDAR_BAUDRATE = 115200;
+const double MM_TO_M = 1000;
+
+std::atomic<bool> lidar_initialized(false);
+std::atomic<datatime_t> lidar_time(datatime_t{});
 RPLidar rp_lidar("/dev/ttyUSB0", RPLIDAR_BAUDRATE);
 std::thread lidar_thread;
-points_t last_points = {};
+DataPoint<points_t> last_points = {};
 std::mutex points_lock;
 
 namespace lidar {
 
-bool initializeLidar() {
-    if (!lidar_initialized) {
-        if (!rp_lidar.checkHealth()) {
-            perror("failed to connect to rp lidar");
-        }
-        rp_lidar.setMaxDistance(16.0);
-        lidar_thread = std::thread(&readLidarLoop);
-    }
-    return lidar_initialized;
-}
-
+/**
+ * @brief RP Lidar continuously scans environment, updates current timeframe with lidar data
+ */
 void readLidarLoop() {
     using namespace std::chrono;
     std::vector<Polar2D> polarPts;    
     while (true) {
         auto scan = rp_lidar.poll();
         if (auto scan = rp_lidar.poll()) {
-            // Converts data into Polar Coord
-            std::vector<Polar2D> currFrames;
+            std::vector<point_t> pts;
             double dtheta = (scan.value().angle_max-scan.value().angle_min)/(scan.value().ranges.size()-1);
             for (unsigned long i = 0; i < scan.value().ranges.size(); i++) {
                 double rad = dtheta*i;
-                double dist = scan.value().ranges[i] * 1000;
+                double dist = scan.value().ranges[i] * MM_TO_M;
 
                 Polar2D frame{dist, rad};
-                currFrames.push_back(frame);
+                pts.push_back(lidar::polarToCartesian2(frame));
             }
         
-            // Converts to regular points
-            std::vector<point_t> pts(currFrames.size());
-            point_t origin({0, 0, 1});
-            for (unsigned long i = 0; i < currFrames.size(); i++) {
-                pts[i] = lidar::polarToCartesian2(currFrames[i]);
-            }
             points_lock.lock();
-            last_points = pts;
+            datatime_t time = lidar_time;
+            last_points = {time, pts};
             points_lock.unlock();
-            lidar_initialized = true;
         } else {
             perror("failed to get frame");
+            points_lock.lock();
+            datatime_t time = lidar_time;
+            last_points = {};
+            points_lock.unlock();
             continue;
         }
     } 
 }
 
+/**
+ * @brief Startsup RPLidar with default settings
+ */
+bool initializeLidar() {
+    if (!lidar_initialized) {
+        if (!rp_lidar.checkHealth()) {
+            perror("failed to connect to rp lidar");
+        } else {
+            rp_lidar.setMaxDistance(16.0);
```

Comment:
Is 16 a hardware limit or is it an arbitrary limiter? If the former this should be a constant at the top, and if the latter then we could either make it a constant or a parameter to `initializeLidar`. Thoughts?

Reply:
When I was testing with the A1 and A2, the limits were around 12m and 16m. I'm all for using the latter since the range for different RP Lidar's ranges, and if were testing with other RP Lidars, we should be able to increase this limit. I'll go ahead and implement an optional parameter for the client.